Automattic/jetpack
# Set WelcomeFlow structure and update WelcomeBanner (async site-only connection)

Related: https://github.com/Automattic/jetpack/pull/36592

## Proposed changes:
- **Introduce WelcomeFlow structure (separate code in preparation to A/B test)**
- **Enhance TermsOfService component**
- **Update MyJetpack connection hook and its usage**
- **Update WelcomeBanner text and add async site registration on click**

![CleanShot 2024-06-05 at 16 29 48](https://github.com/Automattic/jetpack/assets/8419292/8b78c56f-bbcb-4684-8631-6531f5c71914)


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
* Run new Jurassic Ninja site with branch used as Jetpack plugin in Jetpack Beta
* Go to `/wp-admin/admin.php?page=my-jetpack` and notice the welcome banner
* The site activation should happen in the background, and afterwards the banner should disappear
* The site should display as connected

> Note: this PR will be integrated into the A/B test, once they're available.
